src/leap/keymanager/__init__.py

```diff
@@ -185,34 +190,46 @@ def _key_class_from_type(self, ktype):
             lambda klass: klass.__name__ == ktype,
             self._wrapper_map).pop()
 
-    def _get(self, uri, data=None):
+    @defer.inlineCallbacks
+    def _get_json(self, address):
         """
         Send a GET request to C{uri} containing C{data}.
 
         :param uri: The URI of the request.
         :type uri: str
-        :param data: The body of the request.
-        :type data: dict, str or file
 
-        :return: The response to the request.
-        :rtype: requests.Response
+        :return: A deferred that will be fired with the GET response
+        :rtype: Deferred
         """
         leap_assert(
             self._ca_cert_path is not None,
             'We need the CA certificate path!')
```

Comment:
I'm not sure if I agree with you here, but I can let the code pass like that if needed.

This `self._ca_cert_path` is not used at all in this method. If it's somehow set during the life of the KeyManager object it will not afect it as `_async_client_pinned` has this path already set during the init process. I think it could make sense to do this assert in the `__init__` than here.